d2/src/test/java/com/linkedin/d2/xds/TestXdsClientImpl.java

```diff
@@ -331,6 +331,17 @@ public void testHandleD2URICollectionResponseWithData()
     verify(fixture._resourceWatcher).onChanged(eq(expectedUpdate));
     Assert.assertEquals(actualData.getURIMap(), expectedUpdate.getURIMap());
     fixture.verifyAckSent(2);
+
+    // Finally sanity check that the client correctly handles the deletion of the final URI in the collection
+    DiscoveryResponseData deleteUri2 =
+        new DiscoveryResponseData(D2_URI, null, Collections.singletonList(URI_URN2), NONCE, null);
+    fixture._xdsClientImpl.handleResponse(deleteUri2);
+    actualData = (D2URIMapUpdate) fixture._clusterSubscriber.getData();
+    // subscriber data should be updated to D2_URI_MAP_UPDATE_WITH_DATA2
```

Comment:
// subscriber data should be updated to **_empty map_**

Reply:
Done